multivariate_grouped_data_plot.Rmd

```diff
@@ -0,0 +1,825 @@
+# Combine with box plots
+e + geom_boxplot(width = 0.5) + 
+  geom_dotplot(
+    binaxis = "y", stackdir = "center",
+    fill = "white"
+    ) 
+# Dot plot + violin plot + stat summary
+e + geom_violin(trim = FALSE) +
+  geom_dotplot(
+    binaxis='y', stackdir='center',
+    color = "black", fill = "#999999"
+    ) +
+  stat_summary(
+    fun.data="mean_sdl",  fun.args = list(mult=1), 
+    geom = "pointrange", color = "#FC4E07", size = 0.4
+    )
+```
+
+
+#### 3.4.2 创建具有多个组的点图
+
+```{r}
+# Color dots by groups
+e + geom_boxplot(width = 0.5, size = 0.4) +
+  geom_dotplot(
+    aes(fill = supp), trim = FALSE,
+    binaxis='y', stackdir='center'
+  )+
+  scale_fill_manual(values = c("#00AFBB", "#E7B800"))
+# Change the position : interval between dot plot of the same group
+e + geom_boxplot(
+  aes(color = supp), width = 0.5, size = 0.4,
+  position = position_dodge(0.8)
+  ) +
+  geom_dotplot(
+    aes(fill = supp, color = supp), trim = FALSE,
+    binaxis='y', stackdir='center', dotsize = 0.8,
+    position = position_dodge(0.8)
+  )+
+  scale_fill_manual(values = c("#00AFBB", "#E7B800"))+
+  scale_color_manual(values = c("#00AFBB", "#E7B800"))
+```
+
+
+### 3.5 带状图
+
+带状图也称为一维散点图。当样本量较小时，与箱形图相比，带状图更合适。
+
+关键函数：
+
+geom_jitter()。关键参数：
+
+color, fill, size, shape：更改点的颜色，填充，大小和形状
+
+#### 3.5.1 创建一个基本的带状图
+
+- 按组更改点的形状和颜色
+
+- 调整抖动程度：position_jitter(0.2)
+
+- 添加摘要统计信息
+
+```{r}
+e + geom_jitter(
+  aes(shape = dose, color = dose), 
+  position = position_jitter(0.2),
+  size = 1.2
+  ) +
+  stat_summary(
+    aes(color = dose),
+    fun.data="mean_sdl",  fun.args = list(mult=1), 
+    geom = "pointrange",  size = 0.4
+    )+
+  scale_color_manual(values =  c("#00AFBB", "#E7B800", "#FC4E07"))
+```
+
+#### 3.5.2 创建具有多个组的带状图。
+
+R代码类似于在点图部分中看到的代码。
+
+但是，要创建躲避的抖动点，应使用position_jitterdodge()函数而不是position_dodge()。
+
+```{r}
+e + geom_jitter(
+  aes(shape = supp, color = supp), 
+  position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8),
+  size = 1.2
+  ) +
+  stat_summary(
+    aes(color = supp),
+    fun.data="mean_sdl",  fun.args = list(mult=1), 
+    geom = "pointrange",  size = 0.4,
+    position = position_dodge(0.8)
+    )+
+  scale_color_manual(values =  c("#00AFBB", "#E7B800"))
+```
+
+
+### 3.6 Sina图
+
+Sina图的灵感来自于带状图和小提琴图。
+
+通过让标准化的点密度限制沿x轴的抖动，该图显示的轮廓与小提琴图相同，但类似于少量数据点的简单带状图。
+
+这样，该图以非常简单，可理解和精简的格式传达数据点数量，密度分布，离群值和散布信息。
+
+关键函数：
+
+geom_sina（）[ggforce]
+
+```{r}
+library(ggforce)
+# Create some data
+d1 <- data.frame(
+  y = c(rnorm(200, 4, 1), rnorm(200, 5, 2), rnorm(400, 6, 1.5)),
+  group = rep(c("Grp1", "Grp2", "Grp3"), c(200, 200, 400))
+  )
+# Sinaplot
+ggplot(d1, aes(group, y)) +
+  geom_sina(aes(color = group), size = 0.7)+
+  scale_color_manual(values =  c("#00AFBB", "#E7B800", "#FC4E07"))
+```
+
+
+### 3.7 带有误差线的均值和中位数图
+
+在本节中，我们将展示如何绘制由一个或多个分组变量组成的连续变量的摘要统计信息。
+
+请注意，ggpubr软件包提供了一种简单的方法，只需较少的输入即可创建均值/中位数图。 
+
+将默认主题设置为theme_pubr()[在ggpubr中]：
+```{r}
+theme_set(ggpubr::theme_pubr())
+```
+
+#### 3.7.1 基本均值/中位数图
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.